### ISSUE TITLE
docs: write our own tutorials

### ISSUE THREAD

**Issue by JanMa** [openbao/openbao#151]
The current docs contain many links to Vault tutorials in the HashiCorp developer portal. Unfortunately the sources for these tutorials are not public and I don't think it's save for us to just copy the content from the Hashi website and put it into our docs verbatim. 

So instead we should write our own tutorials for at least some topics. I don't think we need to rewrite all ~80 of them, but things like getting started, HA, production hardening and an how to use dynamic DB secrets are definitely needed in my opinion.

**Comment by axelsimon** [comment]
Are you sure?
https://developer.hashicorp.com/vault/docs/what-is-vault has a "[Edit this page on GitHub](https://github.com/hashicorp/vault/blob/main/website/content/docs/what-is-vault.mdx)" link at the bottom which leads to https://github.com/hashicorp/vault/blob/main/website/content/docs/.

One would need to properly check, but if the website and docs were under the same license as the rest of the project, up until https://github.com/hashicorp/vault/pull/22290 on the 11th of August, the docs would fall under the MPL license, meaning the version of the docs at that time could be imported into the OpenBao project.

As usual, IANAL and IANYL etc.

**Comment by cipherboy** [comment]
@axelsimon Yes, that link is for the documentation which is contained under our `website` folder.

Check out this link instead: https://developer.hashicorp.com/vault/tutorials/secrets-management/pki-engine

(see `/tutorials/` in the URL versus `/docs/`)

This was in a different (internal) repo and cannot be edited, e.g., `https://github.com/hashicorp/vault/issues/22761` which discusses this... These tutorials don't have a license associated with them and so likely would have to be rewritten.

**Comment by axelsimon** [comment]
Ah, i see. Thanks for taking the time to clarify!
Indeed, if there is no license associated, then plain copyright applies.
